### PR TITLE
Fix #5329: Add color formatting to static check messages

### LINKED ISSUE #5329: [Feature Request]: Color Formatting Static Checks Messages
### Is your feature request related to a problem? Please describe.

Currently, the output of static checks is presented uniformly in the same color, making it challenging to quickly distinguish failed checks from successful ones. To enhance the experience, color-formatted messages should be implemented for static checks. Specifically, if a static check fails, the corresponding error message should be displayed in red, providing an immediate visual cue for attention while, if the check succeeds, the message should be presented in green, indicating a positive outcome. Warning messages can also be output in yellow for easy scanning.

### Describe the solution you'd like

Shell provides a way to format the color of the output and we can use this to implement the feature. Using color codes, we can be able to implement a printer that takes in a string as a variable and formats it based on the type of message it is. Here is a shell script that can be used as a printer.

```sh
RED='\033[0;31m'
GREEN='\033[0;32m'
YELLOW='\033[0;33m'
NC='\033[0m'

function echo_error() {
    echo -e "${RED}$1${NC}"
}

function echo_success() {
    echo -e "${GREEN}$1${NC}"
}

function echo_warning() {
    echo -e "${YELLOW}$1${NC}"
}
```

Example usage of this printer is as shown below;

```sh
# Import the printer file
source ./path/to/shell_printer.sh

echo_warning "This is a warning message" # Prints a yellow message

echo_error "This is an error message" # Prints a red message

echo_success "This is a success message" # Prints a green message
```

This [blog post](https://ioflood.com/blog/bash-color/) provides more insight on color formatting in shell.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dattasneha**
## Explanation
Fixes #5329  

This PR implements color-coded messages for static checks to improve experience. Error messages are displayed in red for failing checks, while warnings are shown in yellow. Successful checks are indicated with green messages.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by dattasneha** [comment]
@adhiamboperes PTAL.

File: scripts/static_checks.sh
Comment (theMr17) [inline scripts/static_checks.sh:1]: This file contains several unnecessary code changes. Please remove those.

File: scripts/formatting.sh
Comment (theMr17) [inline scripts/formatting.sh:1]: Please remove the empty line at the end of the file. (Note: Every file should end with 1 empty line.)

**Review by theMr17 — CHANGES_REQUESTED** [review]
Thanks @dattasneha! Left a couple of comments, PTAL.

File: scripts/static_checks.sh
Comment (dattasneha) [reply scripts/static_checks.sh:1]: Removed the changes. PTAL.

File: scripts/formatting.sh
Comment (dattasneha) [reply scripts/formatting.sh:1]: Done. @theMr17 PTAL.

**Comment by dattasneha** [comment]
@theMr17, PTAL.

**Review by theMr17 — APPROVED** [review]
Thanks @dattasneha, this LGTM!

File: scripts/formatting.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    echo -e "${RED}$1${NC}"
+}
+
+function echo_success() {
+    echo -e "${GREEN}$1${NC}"
+}
+
+function echo_warning() {
+    echo -e "${YELLOW}$1${NC}"
+}
```
Comment (adhiamboperes) [inline scripts/formatting.sh]: Could you please add comments for future developers? 
```suggestion
# Color definitions
RED='\033[0;31m'    # Red color for errors
GREEN='\033[0;32m'  # Green color for success messages
YELLOW='\033[0;33m' # Yellow color for warnings
NC='\033[0m'        # No Color (reset)

# Function to print error messages in red
function echo_error() {
    echo -e "${RED}$1${NC}"
}

# Function to print success messages in green
function echo_success() {
    echo -e "${GREEN}$1${NC}"
}

# Function to print warning messages in yellow
function echo_warning() {
    echo -e "${YELLOW}$1${NC}"
}
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @dattasneha! Apologies for late review.

In addition to the inline comment, please:
1. Refer to list of static checks [here](https://github.com/oppia/oppia-android/wiki/Static-Analysis-Checks) and update their respective scripts with the new formatting.
2. Update your PR description with some screenshots showing the script in action.

**Comment by dattasneha** [comment]
> Thanks @dattasneha! Apologies for late review.
> 
> In addition to the inline comment, please:
> 
> 1. Refer to list of static checks [here](https://github.com/oppia/oppia-android/wiki/Static-Analysis-Checks) and update their respective scripts with the new formatting.
> 2. Update your PR description with some screenshots showing the script in action.

No issues @adhiamboperes. However I have a doubt regarding the application of bash coloring in .kt files for example in `ReregexPatternValidationCheck.kt` the following code snippet needs to be bash colored. But, I am not sure exactly how to do it.

https://github.com/oppia/oppia-android/blob/eb16e591fcddd369857a86d15f55e649ed1cea17/scripts/src/java/org/oppia/android/scripts/regex/RegexPatternValidationCheck.kt#L63-L67

File: scripts/formatting.sh (hunk 1 above)
Comment (dattasneha) [reply scripts/formatting.sh]: > Could you please add comments for future developers?

Added comments.

**Comment by adhiamboperes** [comment]
@dattasneha, are you experiencig any blockers on this PR?

**Comment by adhiamboperes** [comment]
> No issues @adhiamboperes. However I have a doubt regarding the application of bash coloring in .kt files for example in `ReregexPatternValidationCheck.kt` the following code snippet needs to be bash colored. But, I am not sure exactly how to do it.
> 
> https://github.com/oppia/oppia-android/blob/eb16e591fcddd369857a86d15f55e649ed1cea17/scripts/src/java/org/oppia/android/scripts/regex/RegexPatternValidationCheck.kt#L63-L67

It appears that most terminals already handle exception formatting properly, so we do not need to build a solution for it.

**Review by adhiamboperes — APPROVED** [review]
Thanks @dattasneha! This LGTM.

**Pull request merged**